tests/test_syntax/test_coroutine/test_sdram.py

```diff
@@ -0,0 +1,285 @@
+import tempfile
+import os
+import sys
+
+import magma as m
+import fault
+
+TEST_SYNTAX_PATH = os.path.join(os.path.dirname(__file__), '../')
+
+sys.path.append(TEST_SYNTAX_PATH)
+
+from test_sequential import DefineRegister
+
+
+def test_sdram():
+    CLK_FREQUENCY = 133   # Mhz
+    REFRESH_TIME = 32     # ms     (how often we need to refresh)
+    REFRESH_COUNT = 8192  # cycles (how many refreshes required per refresh time)
+
+    # clk / refresh =  clk / sec
+    #                , sec / refbatch
+    #                , ref / refbatch
+    CYCLES_BETWEEN_REFRESH = \
+        (CLK_FREQUENCY * 1_000 * REFRESH_TIME) // REFRESH_COUNT
+
+    # IDLE = "5'b00000"
+    IDLE = m.bits(0, 5)
+
+    # INIT_NOP1 = "5'b01000"
+    # INIT_PRE1 = "5'b01001"
+    # INIT_NOP1_1 = "5'b00101"
+    # INIT_REF1 = "5'b01010"
+    # INIT_NOP2 = "5'b01011"
+    # INIT_REF2 = "5'b01100"
+    # INIT_NOP3 = "5'b01101"
+    # INIT_LOAD = "5'b01110"
+    # INIT_NOP4 = "5'b01111"
+
+    # REF_PRE = "5'b00001"
+    # REF_NOP1 = "5'b00010"
+    # REF_REF = "5'b00011"
+    # REF_NOP2 = "5'b00100"
+
+    # READ_ACT = "5'b10000"
+    # READ_NOP1 = "5'b10001"
+    # READ_CAS = "5'b10010"
+    # READ_NOP2 = "5'b10011"
+    # READ_READ = "5'b10100"
+
+    # WRIT_ACT = "5'b11000"
+    # WRIT_NOP1 = "5'b11001"
+    # WRIT_CAS = "5'b11010"
+    # WRIT_NOP2 = "5'b11011"
+
+    INIT_NOP1 = m.bits(0b01000, 5)
+    INIT_PRE1 = m.bits(0b01001, 5)
+    INIT_NOP1_1 = m.bits(0b00101, 5)
+    INIT_REF1 = m.bits(0b01010, 5)
+    INIT_NOP2 = m.bits(0b01011, 5)
+    INIT_REF2 = m.bits(0b01100, 5)
+    INIT_NOP3 = m.bits(0b01101, 5)
+    INIT_LOAD = m.bits(0b01110, 5)
+    INIT_NOP4 = m.bits(0b01111, 5)
+    
+    REF_PRE = m.bits(0b00001, 5)
+    REF_NOP1 = m.bits(0b00010, 5)
+    REF_REF = m.bits(0b00011, 5)
+    REF_NOP2 = m.bits(0b00100, 5)
+    
+    READ_ACT = m.bits(0b10000, 5)
+    READ_NOP1 = m.bits(0b10001, 5)
+    READ_CAS = m.bits(0b10010, 5)
+    READ_NOP2 = m.bits(0b10011, 5)
+    READ_READ = m.bits(0b10100, 5)
+    
+    WRIT_ACT = m.bits(0b11000, 5)
+    WRIT_NOP1 = m.bits(0b11001, 5)
+    WRIT_CAS = m.bits(0b11010, 5)
+    WRIT_NOP2 = m.bits(0b11011, 5)
+
+    # CMD_PALL = "8'b10010001"
+    # CMD_REF = "8'b10001000"
+    # CMD_NOP = "8'b10111000"
+    # CMD_MRS = "8'b1000000x"
+    # CMD_BACT = "8'b10011xxx"
+    # CMD_READ = "8'b10101xx1"
+    # CMD_WRIT = "8'b10100xx1"
```

Comment:
Keeping these as reference.  The original verilog has `x` for some of the bits, in magma we don't support this yet, so we just make them 0 for now.